src/libsyncengine/update_detection/update_detector/node.cpp

```diff
@@ -180,16 +181,25 @@ SyncPath Node::getPath() const {
     return path;
 }
 
-bool Node::isParentValid(std::shared_ptr<const Node> parentNode) const {
-    if (!parentNode) return true; // `parentNode` is the root node, hence a valid parent. Stop climbing up the tree.
-
-    while (parentNode) {
-        if (parentNode->id() == _id)
-            return false; // This node is a parent of `parentNode`, hence `parentNode` is not a valid parent.
-        parentNode = parentNode->parentNode();
+bool Node::isParentOf(std::shared_ptr<const Node> potentialChild) const {
+    if (!potentialChild) return false; // `parentNode` is the root node, 
+    if (potentialChild->id().has_value() && potentialChild->id() == _id) return false; // `parentNode` is the root node
```

Comment:
Is the comment accurate?

Reply:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/356/commits/da844e86f7878e113e9f0e9c0bfec3aed38cb946